pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -0,0 +1,255 @@
+package guard
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/informers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	policyclientv1 "k8s.io/client-go/kubernetes/typed/policy/v1"
+	corelisterv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1listers "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/guard/bindata"
+)
+
+// GuardController is a controller that watches amount of static pods on master nodes and
+// renders guard pods with a pdb to keep maxUnavailable to be at most 1
+type GuardController struct {
+	targetNamespace, podResourcePrefix string
+	operatorName                       string
+	readyzPort                         string
+
+	nodeLister corelisterv1.NodeLister
+	podGetter  corev1client.PodsGetter
+	pdbGetter  policyclientv1.PodDisruptionBudgetsGetter
+
+	// installerPodImageFn returns the image name for the installer pod
+	installerPodImageFn   func() string
+	createConditionalFunc resourceapply.ConditionalFunction
+	deleteConditionalFunc resourceapply.ConditionalFunction
+}
+
+func NewGuardController(
+	targetNamespace, podResourcePrefix string,
+	operatorName string,
+	readyzPort string,
+	kubeInformersForTargetNamespace informers.SharedInformerFactory,
+	kubeInformersClusterScoped informers.SharedInformerFactory,
+	podGetter corev1client.PodsGetter,
+	pdbGetter policyclientv1.PodDisruptionBudgetsGetter,
```

Comment:
is this passed a caching GET client?

Reply:
The same that's passed to other staticpod controllers